Rakefile

```
@@ -18,3 +18,21 @@ else
     config.future_release = Gem::Specification.load("#{config.project}.gemspec").version
   end
 end
+
+begin
+  require 'rubocop/rake_task'
+rescue LoadError
+  # RuboCop is an optional group
+else
+  RuboCop::RakeTask.new(:rubocop) do |task|
+    # These make the rubocop experience maybe slightly less terrible
+    task.options = ['--display-cop-names', '--display-style-guide', '--extra-details']
+    # Use Rubocop's Github Actions formatter if possible
+    if ENV['GITHUB_ACTIONS'] == 'true'
+      rubocop_spec = Gem::Specification.find_by_name('rubocop')
+      if Gem::Version.new(rubocop_spec.version) >= Gem::Version.new('1.2')
+        task.formatters << 'github'
+      end
```

You ensure a new enough version in the dev dependency spec, so this is fine.
```suggestion
      task.formatters << 'github'
```